rds/aurora/mysql/cluster/rds.go

```diff
@@ -69,6 +69,10 @@ func NewRDSStack(scope constructs.Construct, props *rds.RDSStackProps) awscdk.St
 		}),
 	})
 
+	if props.RestoreSnapshotArn != "" {
+		awscdk.Aspects_Of(cluster).Add(rds.NewSnapshotAspect(props.RestoreSnapshotArn))
+	}
```

Comment:
This is all that is needed for the restoration to happen? It won't happen again during the next run?

Reply:
The Aspect_Of just sets the value on the cluster object. It traverses all the nodes looking for a specific one. If I didn't do this, and still wanted to use the higher CDK constructs I would have needed a way to pick a path between `awsrds.NewDatabaseCluster` and `awsrds.NewDatabaseClusterFromSnapshot` which have different types. 

It actually needs to happen every time. Cloudformation is smart enough to know once it has done the restore to ignore it on subsequent runs. If it is changed then it initiates another restore. That is why the hooks have been added, so if the user blanks it out to the empty string they don't end up with an empty db. Though the user should be able to set it to multiple restore points over the life of the Acorn.